### PR TITLE
add-flipkart-price-alert-with-changes

### LINKED ISSUE #541: Flipkart Price Alert
# Aim
I am a participant in GSSOC'21.
As the title says , the script checks for the price of a desired product periodically. If the price drops below the desired amount, notifies the user via email.

# Details
Will be using BeautifulSoup to scrape webpages.

### PR DESCRIPTION
**Pull request by Jade9ja**
# Description

Added the flipkart price alert script.

## Fixes #541 

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [X] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [X] My code follows the style guidelines(Clean Code) of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have created a helpful and easy to understand `README.md`
- [X] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [X] My changes generate no new warnings
- [X] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Flipkart-price-alert/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Flipkart Price Alert
+Checks for the price of a desired product periodically.
+If the price drops below the desired amount, notifies the user via email.
+# Setup Instructions
+Type the following to install requirements: (make sure you have python installed)
+`pip install -r requirements.txt`
+This script requires access to mail credentials inorder to work, follow the steps below to generate the credentials for a limited scope.
+## How to get app password
```
Comment (kaustubhgupta) [inline Flipkart-price-alert/README.md:8]: It's confusing to write "app passwords". I would suggest rewriting it as "Optional: How to get alternate login credentials"

File: Flipkart-price-alert/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Flipkart Price Alert
+Checks for the price of a desired product periodically.
+If the price drops below the desired amount, notifies the user via email.
+# Setup Instructions
```
Comment (kaustubhgupta) [inline Flipkart-price-alert/README.md:4]: All heading levels after the "# Flipkart Price alert" heading need to be on the same level. Either make them (##) or (#)

File: Flipkart-price-alert/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## How to get app password
+1) To get app passwords, go to myaccount.google.com/security 
+2) Make sure you save 2-step verification turned on.
+3) Choose app passwords (under signing into google)
+4) Then under app choose mail and under platform, choose the device from which you are logged in from.
+
+![](https://devanswers.co/wp-content/uploads/2017/02/my-google-app-passwords.png)
+
+## Usage
+1) Edit the config vars in the track.py script.
```
Comment (kaustubhgupta) [inline Flipkart-price-alert/README.md:17]: This line is not needed any more

File: Flipkart-price-alert/track.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    # If the price falls below threshold, send an email
+    if(price < THRESHHOLD):
+        sendMail(title)
+        
+        
+        
+PRODUCT_URL = input('Enter the product url:')
+THRESHHOLD = float(input('Enter the desired price:'))
+MY_EMAIL = input('Enter the email address that will be used to send alerts:')
+MY_APP_PASSWORD = input('Enter the app password (Check readme for steps to generate app passwords):')
```
Comment (kaustubhgupta) [inline Flipkart-price-alert/track.py:46]:
```suggestion
MY_APP_PASSWORD = input('Enter password:')
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
The code is well written, just a few minor changes:

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@antrikshmisri @santushtisharma10

**Comment by santushtisharma10** [comment]
LGTM 💯

**Review by santushtisharma10 — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]
LGTM 🚀

**Review by avinashkranjan — APPROVED** [review]
LGTM..

**Pull request merged**